File: 프로그래머스/0/120852. 소인수분해/소인수분해.js
function solution(n) {
  let k = 2;
  let arr = []

  while (n != 1) {
    if (n % k == 0) {
      arr.includes(k) ? arr : arr.push(k)
      n /= k;
    } else {
      k++;
    }
  }
  return arr
}

File: 프로그래머스/0/120852. 소인수분해/소인수분해.java
import java.util.ArrayList;

class Solution {
    public int[] solution(int n) {
        int k = 2;
        ArrayList<Integer> arr = new ArrayList<>();
        
        while (n != 1) {
            if(n % k ==0) {
                if(!arr.contains(k)) {
                    arr.add(k);
                }
                n /= k;
            } else {
                k++;
            }
        }
        
        int[] result = new int[arr.size()];
        for (int i = 0; i < arr.size(); i++) {
            result[i] = arr.get(i);
        }
        
        return result;
    }
}